### ISSUE TITLE
struct timeval is not declared

### ISSUE THREAD

**Issue by ef502f79-e8ca-42bb-a095-2e48c43425bb** [python/cpython#82863]
BPO | [38682](https://bugs.python.org/issue38682)
--- | :---
Nosy | @pfmoore, @tjguk, @zware, @zooba, @mithil467, @iritkatriel

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2022-01-29.20:02:00.670>
created_at = <Date 2019-11-04.09:36:12.418>
labels = ['build', '3.7', 'OS-windows']
title = 'struct timeval is not declared'
updated_at = <Date 2022-01-29.20:02:00.669>
user = 'https://github.com/mithil467'
```

bugs.python.org fields:
```python
activity = <Date 2022-01-29.20:02:00.669>
actor = 'iritkatriel'
assignee = 'none'
closed = True
closed_date = <Date 2022-01-29.20:02:00.670>
closer = 'iritkatriel'
components = ['Windows']
creation = <Date 2019-11-04.09:36:12.418>
creator = 'Mithil'
dependencies = []
files = []
hgrepos = []
issue_num = 38682
keywords = []
message_count = 2.0
messages = ['355932', '412101']
nosy_count = 6.0
nosy_names = ['paul.moore', 'tim.golden', 'zach.ware', 'steve.dower', 'Mithil', 'iritkatriel']
pr_nums = []
priority = 'normal'
resolution = 'out of date'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'compile error'
url = 'https://bugs.python.org/issue38682'
versions = ['Python 3.7']
```

</p></details>

**Comment by mattip** [comment]
fwiw, this was fixed for clang in #77532, which became part of 3.12

**Comment by sfinktah** [comment]
@mattip fwiw, it was worth a lot.

# For anyone being annoyed by this warning in py <= 3.11

Add this at ~line 55 of include/cpython/pytime.h (where-ever good python installs are kept).

```cpp
#ifdef __clang__
struct timeval;
#endif
```

Actually, this being a Windows+clang specific problem, good pythons are kept at `%USERPROFILE%/AppData/Local/Programs/Python/Python311/include/cpython/pytime.h`

(everyone else, it was fixed in py 3.12)
